### ISSUE TITLE
Error with lead SNP file, and plotting

### ISSUE THREAD

**Issue by johnshorter** [comorment/containers#22]
There is an error in the loci command. The output from loci is used when plotting.
This is the description of the loci command:
Perform LD-based clumping of summary stats, using a procedure that is similar to FUMA snp2gene functionality
The plotting seems to expect the files iPSYCH2012_ind_F3300.lead.csv and iPSYCH2012_ind_F3300.indep.csv, but those files does not exist. I guess they would be produced by the loci command

I tried to re-run and remove the flags for those two files to see if that solves the issue. But it didn’t. And I know that chr 2 has variants that pass significance threshold. 

![image](https://user-images.githubusercontent.com/19779071/138437023-bdac0ba6-a8c7-4550-a5c4-687b3bd39c5c.png)

**Comment by ofrei** [comment]
@johnshorter hi, I think a quick solution  is to remove ``loci`` from ``gwas.py --analysis`` argument, could you try this?
Even if this works, please keep this ticket open - I'd like to make another version of ``loci`` that uses 1MB window to clump based on distance, without requiring a reference panel for LD pruning.

**Comment by johnshorter** [comment]
This is the error when I run the manh plot. It seems to exclude all markers. 

![image](https://user-images.githubusercontent.com/19779071/140066192-42b02fcd-202c-4c92-a494-eabdabf9c446.png)

**Comment by ofrei** [comment]
@johnshorter The SNPs were removed due to ``--snps-visual-bins`` argument ( https://github.com/comorment/containers/blob/main/gwas/gwas.py#L267 ), which  bins SNPs into 1000 bins both horizontally and vertically so that worst case it's 1.000.000 points (in reality a lot less as there is plenty of empty space on Manhattan plot). So the last line isn't an error or warning - but I agree it's quite misleading. I've changed the script to output total number of SNPs:
https://github.com/comorment/containers/commit/75ddfb5a891bfeac07a436e708018d0e99c1cc80
 
 Could you double-check whether the actual  manhattan plot was produced and looked reasonable?

**Comment by johnshorter** [comment]
The job just stalls there, and there is no actual manhattan plot made. It is now at 4 hours, and will stop at my 6 hour limit and not do anything more.

**Comment by ofrei** [comment]
@johnshorter could you copy 
- your "gwas.py" command that generates all analysis scripts, and
- the shell script that failed with the above error once you've submitted it to sbatch

**Comment by johnshorter** [comment]
![image](https://user-images.githubusercontent.com/19779071/140303448-e9f6bc1c-b95c-4ec7-b553-2658d4b81265.png)
This is before I removed the "loci" from the gwas.py argument. I think it is necessary in order to run because of this error. 
![image](https://user-images.githubusercontent.com/19779071/140303706-6efbac45-6eb5-4374-accb-a38330325992.png)

This was the last command and error I did before trying to remove "loci"

![image](https://user-images.githubusercontent.com/19779071/140304159-fe766f6a-5010-41cf-90da-0a34b6d4252e.png)

and the bottom 

![image](https://user-images.githubusercontent.com/19779071/140304223-85ab26ad-74cc-4f3e-addc-b159b7c6e30a.png)

**Comment by ofrei** [comment]
@johnshorter Sorry for the trouble - I really couldn't reproduce this on my end. There are few things we can do to investigate:

- one of the logs has ``module: command not found error``. This looks suspicious to me, could you check where this is coming from? The only time this enters my scripts is via ``module load singularity``, perhaps this need to be done separately in your HPC environment? Or it could be a harmless message, I don't know.

- your ``gwas.py manh`` command still has ``--lead`` and ``-indep`` arguments pointing to the output of ``gwas.py loci``. Please re-run ``gwas.py gwas`` step with ``--analysis`` argument that doesn't include ``loci``. This way the script for manhattan plot will be re-generated, and it won't have ``--lead`` and `--indep`` arguments. Alternatively, just edit the ``.job`` script (in my case it was ``run1_plink2.3.job`` script) to delete ``--lead`` and ``--indep`` argument from ``gwas.py manh`` command. 

- could you copy here the full script containing ``gwas.py manh`` command (i.e. ``run1_plink2.3.job`` script), and the full SLURM output (what happens if you execute that script)? Manhattan plot should be built in a matter of seconds - so in principle you can even run in on a login node without SLURM. 

If the above doesn't help we can have a short chat on Zoom to figure this out.

**Comment by johnshorter** [comment]
![image](https://user-images.githubusercontent.com/19779071/140328585-6042bb47-c3b0-4627-b664-57ed52dcc44b.png)
Here is my edited script for job 4. 

here is a strange error when I run it
![image](https://user-images.githubusercontent.com/19779071/140328689-2d2b79b0-4332-4d11-97e7-427dbf733d23.png)


And then no manh plot is ever made, it will just keep going, and will time out after 6 hours
![image](https://user-images.githubusercontent.com/19779071/140328840-bd1bc8a8-d302-44e6-80af-6ed287a6c8b5.png)

**Comment by johnshorter** [comment]
![image](https://user-images.githubusercontent.com/19779071/140505665-07f46add-6823-45a6-979c-a373dbe37180.png)

**Comment by ofrei** [comment]
this is now implemented via new ``gwas.py --analysis figures`` option. It generates manhattan and qq plots using ``qqman`` package. Full command looks as follows:
``
$RSCRIPT -e 'library(data.table);library(qqman);library(ggplot2);df=read.table("run3_saige_CASE2.gz", header=TRUE);png("run3_saige_CASE2.qq.png", width=600, unit="px", pointsize=12, bg="white");qq(df$P, main="CASE2");dev.off();df=df[df$P < 0.05, ];png("run3_saige_CASE2.manh.png", width=1200, unit="px", pointsize=12, bg="white");manhattan(df, chr="CHR", bp="BP", snp="SNP", p="P", main="CASE2");dev.off();'
`` (the only trick  here is to filter p>0.05 on manhattan plot, to make it more lightweight).

I marked previous visualization options (``loci``, ``manh`` and ``qq``)  as deprecated. it's best to stick to  ``qqman`` as it is more robust. As for ``loci``, there is no single definition of a locus that works 100% of the time. For now it's best to let each project discuss and define it's locus definition - no need to have this as part of gwas.py pipeline.
